rest/route/annotations.go

```diff
@@ -265,8 +266,7 @@ func (h *annotationByTaskPutHandler) Parse(ctx context.Context, r *http.Request)
 	}
 
 	// check if the task exists
-	// t, err := task.FindOne(task.ById(h.taskId))
-	t, err := task.FindByIdExecution(h.taskId, h.annotation.TaskExecution)
+	t, err := task.FindOne(task.ById(h.taskId))
```

Comment:
We need this to check both. This fixes a bug where `if !evergreen.IsFailedTaskStatus(t.Status) ` on line 279 doesn't allow for annotation to be filed if a task fails for execution 0 but succeeds for execution 1.

Reply:
Added back to find by execution